pkg/athena/driver/rows.go

```diff
@@ -41,27 +40,27 @@ func (r *Rows) ColumnTypeScanType(index int) reflect.Type {
 
 func (r *Rows) athenaTypeOf(columnInfo *athena.ColumnInfo) (reflect.Type, error) {
 	switch *columnInfo.Type {
-	case "tinyint":
-		return reflect.TypeOf(int8(0)), nil
-	case "smallint":
-		return reflect.TypeOf(int16(0)), nil
+	case "tinyint", "smallint":
+		return reflect.TypeOf(sql.NullInt16{}), nil
 	case "integer":
-		return reflect.TypeOf(int32(0)), nil
+		return reflect.TypeOf(sql.NullInt32{}), nil
 	case "bigint":
-		return reflect.TypeOf(int64(0)), nil
-	case "float", "real":
-		return reflect.TypeOf(float32(0)), nil
-	case "double":
-		return reflect.TypeOf(float64(0)), nil
+		return reflect.TypeOf(sql.NullInt64{}), nil
+	case "float", "real", "double":
+		return reflect.TypeOf(sql.NullFloat64{}), nil
 	case "json", "char", "varchar", "varbinary", "row", "string", "binary",
 		"struct", "interval year to month", "interval day to second", "decimal",
 		"ipaddress", "array", "map", "unknown":
-		return reflect.TypeOf(""), nil
+		return reflect.TypeOf(sql.NullString{}), nil
 	case "boolean":
-		return reflect.TypeOf(false), nil
+		return reflect.TypeOf(sql.NullBool{}), nil
 	case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-		return reflect.TypeOf(time.Time{}), nil
+		return reflect.TypeOf(sql.NullTime{}), nil
 	default:
 		return nil, fmt.Errorf("unknown column type `%s`", *columnInfo.Type)
 	}
 }
+
+func (r *Rows) ColumnTypeNullable(index int) (nullable, ok bool) {
```

Comment:
Where is this function used? I can't find it referenced 🤔

Reply:
This is called inside the go sql code. https://github.com/golang/go/blob/f81c885819cfca1b92ee64abb0b932e4f73b5049/src/database/sql/sql.go#L3170